opensoar/competition/soaringspot.py

```diff
@@ -358,8 +382,25 @@ def generate_competition_day(self, target_directory: str, download_progress=None
                 print('{} is skipped because of invalid trace'.format(competition_id))
                 continue
 
+            # Remove double time entries. Remove the latest one if double entries exist.
+            for index_right, index_left in double_iterator(range(len(trace) - 1, -1, -1)):  # Reversed order
```

Comment:
i have started the conversation on removing duplicate fixes in the aerofiles library: https://github.com/Turbo87/aerofiles/pull/298

Reply:
also started a conversation on the timezone conversion: https://github.com/Turbo87/aerofiles/issues/299. interestingly enough the IGC file also contains a date, i was not aware of that (which is why i took it from soaringspot instead)

Reply:
The aerofiles maintainer is ok with adding it there, which is a better solution i think. So lets move the logic there.

Reply:
Leaving this in for now. Did reduce the complexity of the code

Reply:
i don't really follow what you mean by "Leaving this in for now". are you planning to remove it once it's in the aerofiles library? because i believe that's the way forward. 

in the meantime i will check if i can figure out if the tests are failing for another reason

Reply:
Picked up in https://github.com/Turbo87/aerofiles/pull/304 ! So this loop here can be removed once merged on aerofiles

Reply:
upgrade aerofiles dependency and removed this block, also pinned all other dependencies